packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -267,26 +273,29 @@ export default (superClass: Class<Parser>): Class<Parser> =>
           } else {
             modified.accessibility = modifier;
           }
+          enforceOrder(startPos, modifier, "override");
         } else {
           if (Object.hasOwnProperty.call(modified, modifier)) {
             this.raise(startPos, TSErrors.DuplicateModifier, modifier);
-          } else if (modified.readonly && modifier === "static") {
-            this.raise(
-              startPos,
-              TSErrors.InvalidModifiersOrder,
-              "static",
-              "readonly",
-            );
-          } else if (
-            (modified.declare && modifier === "override") ||
-            (modified.override && modifier === "declare")
-          ) {
-            this.raise(
-              startPos,
-              TSErrors.IncompatibleModifiers,
-              "declare",
-              "override",
-            );
+          } else {
+            if (modifier === "static") {
+              enforceOrder(startPos, modifier, "readonly");
+              enforceOrder(startPos, modifier, "override");
+            } else if (modifier === "override") {
+              enforceOrder(startPos, modifier, "readonly");
+            }
+
+            if (
+              (modified.declare && modifier === "override") ||
+              (modified.override && modifier === "declare")
+            ) {
+              this.raise(
+                startPos,
+                TSErrors.IncompatibleModifiers,
+                "declare",
+                "override",
+              );
+            }
```

Comment:
`static` can not be used with `abstract`: [TS Playground](https://www.typescriptlang.org/play?ts=4.3.0-dev.20210426&ssl=1&ssc=1&pln=8&pc=1#code/IYIwzgLgTsDGEAJYBthjAgQmgpgYVXQQG8AoBBSYCAS1gQFsBGALgWADsBPUgX1NKhIMeEkIYAslwJowAVgQ4AHhBwcAJhmxh84kuUoRqdduGhxEzNsJocA5gG4+pIA).

Reply:
Oh it looks like we didn't have any test for it